src/cascadia/QueryExtension/AzureLLMProvider.cpp

```diff
@@ -145,7 +153,9 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
             // Send the request
             try
             {
-                const auto response = _httpClient.SendRequestAsync(request).get();
+                const auto sendRequestOperation = _httpClient.SendRequestAsync(request);
+                const auto response{ co_await sendRequestOperation };
+                _lastRequest = sendRequestOperation;
```

Comment:
I don't think this is effective if you set it _after_ the wait on the request. The `.get()` below should IMO also be replaced with a `_lastRequest = ...` and `co_await`.